### PR TITLE
Analysis API + New map config format (includes analyses)

### LINKED ISSUE #1170: Analysis API
TBD: How should the editor send the actual state of layers/analysis/widgets to CartoDB.js? (One change at a time VS sending the state "in bulk")

cc: @viddo @javierarce @xavijam

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
This PR fixes #1170 and #1171 (and some other PRs made against `analysis`). This PR is getting too big so I think it's a good idea to merge it ASAP. Some highlights:

- Introduces API that is being used by the editor to create analysis (`vis.analyses.analize`, setting `source` attribute in layers, refreshing map when analyses related stuff changes, etc.).
- Refactoring to make generation of Map Config simpler (check out `src/windshaft/anonymous-map.js
` and `src/windshaft/named-map.js`).
- Map Config for anonymous maps now includes `analyses` when present (it'll continue working as before otherwise).
- Changes the way dataviews and filters are sent to Maps API (dataviews were included as `widgets` in the layer `options` and are now a separate `dataviews` entry in the Map config and filters where grouped by layer).
- `cartodb.createVis` accepts `apiKey` option. If present, this API key is used for all requests to the Maps API (map instantiation, tiles & grids, dataviews, attributes...).

@xavijam sorry in advance for the huge PR... could you please take a look? Thanks!

### CODE REVIEW

**Comment by alonsogarciapablo** [comment]
retest this please

**Comment by alonsogarciapablo** [comment]
@xavijam I didn't realise this PR will "break" deep-insights.js (because of the change on how dataviews/filters are sent to the Maps API), so we'll have to wait until the Maps API release.

**Comment by xavijam** [comment]
ok :+1:

File: .eslintignore [hunk 1]
```diff
@@ -50,7 +50,6 @@
 **/src/geo/leaflet/leaflet-point-view.js
 **/src/geo/leaflet/leaflet-tiled-layer-view.js
 **/src/geo/leaflet/leaflet-wms-layer-view.js
-**/src/geo/map/cartodb-layer.js
```
Comment (xavijam) [inline .eslintignore]: not cartodb-layer anymore?

**Comment by xavijam** [comment]
👍

File: .eslintignore (hunk 1 above)
Comment (alonsogarciapablo) [inline .eslintignore]: That file has no lint errors anymore, that's why we're removing it from this file.

**Pull request merged**